### PR TITLE
Initial commit to go mod

### PR DESCRIPTION
**Pull request by notzippy**
This is a WIP for upgrading the command line tool to use go modules by default
What is done

* Create new project, you can specify the package name as well. 
* `go.mod` files are created in the base of the package source code and can be found by the revel run command
* The function ProcessSource [old](https://github.com/notzippy/revel-cmd/blob/go-mod/parser/reflect.go#L37) [new](https://github.com/notzippy/revel-cmd/blob/go-mod/parser2/read.go#L13) has been started to be rewritten to use the go module friendly reflection tool at `golang.org/x/tools/go/package`. 

ToDo
* Follow the code for the [old](https://github.com/notzippy/revel-cmd/blob/go-mod/parser/reflect.go#L37) and add that into the [new](https://github.com/notzippy/revel-cmd/blob/go-mod/parser2/read.go#L13). 

The goal is to completely rewrite the `ProcessSource` function, and we will eventually drop the old implementation

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module github.com/revel/cmd
+
+go 1.13
```
Comment (ptman) [inline go.mod:3]: Shouldn't this be 1.12 if we try to build using 1.12?

File: harness/app.go [hunk 2]
```diff
@@ -64,17 +65,17 @@ func (cmd AppCmd) Start(c *model.CommandConfig) error {
...
 		utils.Logger.Fatal("Error running:", "error", err)
 	}
 
 	select {
 	case exitState := <-cmd.waitChan():
 		fmt.Println("Startup failure view previous messages, \n Proxy is listening :", c.Run.Port)
-		err := utils.NewError("","Revel Run Error", "starting your application there was an exception. See terminal output, " + exitState,"")
-		// TODO pretiffy command line output
-		// err.MetaError = listeningWriter.getLastOutput()
+		err := utils.NewError("", "Revel Run Error", "starting your application there was an exception. See terminal output, " + exitState, "")
```
Comment (ptman) [inline harness/app.go:76]: Are you running gofmt / goimports on this code? Most editors support hooking it up.

File: harness/build.go [hunk 3]
```diff
@@ -135,30 +144,43 @@ func Build(c *model.CommandConfig, paths *model.RevelContainer) (_ *App, err err
...
 
+	for {
 		appVersion := getAppVersion(paths)
 		if appVersion == "" {
 			appVersion = "noVersionProvided"
 		}
 
 		buildTime := time.Now().UTC().Format(time.RFC3339)
-		versionLinkerFlags := fmt.Sprintf("-X '%s/app.AppVersion=%s' -X '%s/app.BuildTime=%s'",
+		versionLinkerFlags := fmt.Sprintf("-X %s/app.AppVersion=%s -X %s/app.BuildTime=%s",
```
Comment (ptman) [inline harness/build.go:168]: isn't this good if the path contains spaces?

File: harness/build.go [hunk 4]
```diff
@@ -135,30 +144,43 @@ func Build(c *model.CommandConfig, paths *model.RevelContainer) (_ *App, err err
...
 			paths.ImportPath, appVersion, paths.ImportPath, buildTime)
 
 		// Append any build flags specified, they will override existing flags
 		flags := []string{}
 		if len(c.BuildFlags) == 0 {
 			flags = []string{
 				"build",
-				"-i",
```
Comment (ptman) [inline harness/build.go:154]: Great!

File: harness/app.go (hunk 2 above)
Comment (notzippy) [reply harness/app.go:76]: When I touch a file I run the gofmt, goimports. But there is a lot of code that is not formatted properly from start. I will do a separate houseclean pr on a clean develop branch for all project files in the near future

File: harness/build.go (hunk 3 above)
Comment (notzippy) [reply harness/build.go]: Yup my fault, bad merge, reverted

File: go.mod (hunk 1 above)
Comment (notzippy) [reply go.mod]: Fixed

**Pull request merged**